include/deal.II/base/tensor_function.templates.h

```diff
@@ -140,10 +139,8 @@ ConstantTensorFunction<rank, dim, Number>::gradient_list(
   Assert(gradients.size() == points.size(),
          ExcDimensionMismatch(gradients.size(), points.size()));
 
-  static const Tensor<rank + 1, dim, Number> zero{};
-
   for (unsigned int i = 0; i < gradients.size(); ++i)
-    gradients[i] = zero;
+    gradients[i] = gradient_type();
```

Comment:
Again, you need the full name `typename TensorFunction<rank, dim, Number>::gradient_type`

Reply:
It's probably best to use `std:fill` here.